app/src/main/java/org/metabrainz/mobile/repository/LookupRepository.java

```diff
@@ -118,21 +115,45 @@ public void onFailure(Call<ResponseBody> call, Throwable t) {
         });
     }
 
-    public MutableLiveData<List<Release>> fetchCoverArt(List<Release> releases, int position){
-        Release release = releases.get(position);
-        service.getCoverArt(release.getMbid()).enqueue(new Callback<CoverArt>() {
+    /**
+     * For a given release ID, fetches the cover arts and updates the release w¡th that info
+     * @param releaseId MusicBrainz ID of the release.
+     * @param position Release item position in the artsit's releases array, to update it.
+     * @return
+     */
+    public void fetchCoverArtForRelease(String releaseId, int position){
+        service.getCoverArt(releaseId).enqueue(new Callback<CoverArt>() {
             @Override
             public void onResponse(Call<CoverArt> call, Response<CoverArt> response) {
-                CoverArt art = response.body();
-                release.setCoverArt(art);
-                coverArtMutableLiveData.setValue(releases);
+                if (response.code() == 200) {
+                    // Only found cover arts (200 OK) must be used
+                    CoverArt coverArt = response.body();
+
+                    // Add coverArt to artist's release.
+                    Artist artist = artistData.getValue();
+                    if (artist != null) {
+                        ArrayList<Release> releases = artist.getReleases();
+
+                        if (releases != null) {
+                            Release release = releases.get(position);
+
+                            // Replace cover art for this release
+                            release.setCoverArt(coverArt);
+
+                            // Update artist release list
+                            artist.setRelease(release, position);
+
+                            // Resend the LiveData for any observer to get the new cover art
+                            artistData.postValue(artist);
```

Comment:
Brilliant idea to use only artist but I sense some complications. For instance, every time the artist is updated with a cover art, it will notify other observers as well that is to say that, `setArtist` , `getArtist` etc. will get called as well, which is unnecessary processing. I'm not sure though if I am right.